gson/src/main/java/com/google/gson/JsonSchemaValidator.java

```diff
@@ -0,0 +1,259 @@
+package com.google.gson;
+
+import java.net.URI;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.function.Function;
+
+public abstract class JsonSchemaValidator {
+    public static void validate(JsonElement schema) throws SchemaValidationException {
+        if (validateSchemaRootType(schema)) return;
+        JsonObject schemaRoot = schema.getAsJsonObject();
+        verifyTypeField(schemaRoot, MemberTypes.OBJECT);
+        validateSchemaNode(schema);
+    }
+
+    static boolean validateSchemaRootType(JsonElement schema) {
+        boolean isObject = schema.isJsonObject();
+        if (!isObject){
+            // if schema root is not an Object, it must be a boolean
+            schema.getAsBoolean();
+            return true;
```

Comment:
I think the schema.getAsBoolean() call will throw an exception if the root is not a `Boolean`.

Consider changing this to:

```
if (schema.isJsonPrimitive() && schema.getAsJsonPrimitive().isBoolean())
    return true;
```